File: rmaregress.py
import numpy as np
import scipy.stats as st 


def rmaregress(x,y):
    slope, inter, r, p, std = st.linregress(x, y)
    rma_slope = slope/r
    rma_inter = np.mean(y)-np.mean(x)*rma_slope  
    rma_p = p 
    rma_R2 = r**2
    return rma_slope, rma_inter, rma_R2, rma_p 

File: rmaregress.m
function [b,R2,p]=rmaregress(x,y)
x=x(:);y=y(:);

% Remove missing values, if any
wasnan = (isnan(y) | any(isnan(x),2));
havenans = any(wasnan);
if havenans
   y(wasnan) = [];
   x(wasnan,:) = [];
   %n = length(y);
end
X = [ones(size(y)),x];
[b1, ~,~,~,stats]=regress(y,X);

b=zeros(2,1);
b(2)=b1(2)/stats(1)^0.5;% slope
b(1)=mean(y)-mean(x)*b(2);%interception

R2=stats(1);
p=stats(3);

end